### DISCUSSION TITLE
I can bypass my hotspot data cap, but the connection is very slow

### DISCUSSION THREAD

**Discussion by yantlord** [Mygod/VPNHotspot#334] [Q&A] [answered]
I'm using the app on a Moto G Fast. I've disabled hardware acceleration for tethering, also have IPv6 disabled. I'm able to throw the sim into an older device and the connection is not throttled, so I know it's possible within the network. Fuck this new tech that snitches to the carrier, I'd really like to get this working just out of spite at this point.

Any ideas? I was unable to figure out how to use App Systemizer, is that making the difference? It seems like the project is dead and no longer functional with new versions of Android, I did get it installed in Magisk but that's as far as I got.

Anyway, thanks for the good work homie, the app looks great and it does work, I just wish I could get that data flowing a bit more freely, you know what I mean? Let me know if you have any suggestions

Thanks :)

**Comment by Mygod** [comment, marked as answer]
Which VPN were you using? If you are using a fully fledged VPN, you might want to try something lighter and see if it improves performance. See https://github.com/Mygod/VPNHotspot/discussions/216.

**Reply by yantlord** [reply]
Shit sorry, didn't realize I should have replied here to begin with... Anyway, just wanted to say thanks again homie, I got it working! I'm still running the PIA VPN app too, getting normal speeds. Gonna keep playing around and see if I can get a bit more out of it by trying some of the suggestions in lieu of using the PIA app. I was super blazed last time I was trying to get this going, and I didn't realize I needed to manually flip the interface switch on after activating the hotspot in the app... :S

Keep up the good work, you're helping a lot of people. I'll put some loot through soon, my whole family really appreciates everything you've done here!

**Comment by yantlord** [comment]
Thanks for the reply :)

I'm using PIA and their app. As I understand it, you're saying that the PIA app may be too resource-intensive to work in conjunction with VPNHotspot. I took a quick look at the link (will read more thoroughly later) and it looks like the options there are either some sort of decoy ad blocker (?) or else try WireGuard. The PIA app has a protocol selection to use their implementation of WireGuard, I will try that first and see if it makes a difference in performance. The PIA app is convenient but it's pretty bogged down just navigating between menus, so you're probably correct that this is what's jamming me up. 

I also noticed when checking which protocol is selected, that the transport mechanism is set to UDP. That shouldn't make a difference since everything is being tunneled, right?

I also switched the "Use small packets" option on and will see if that makes any difference.

Finally - if the goal is to lighten the load, do you know if anyone has tried using their VPN provider config within the Android system VPN options vs using the VPN provider's proprietary apps?

**Reply by Mygod** [reply]
WireGuard might still be pretty heavy compared to VpnService-based ad blockers.
